src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java

```diff
@@ -335,18 +336,28 @@ private void updateIgnoredWordsIndex()
 
    private void loadDictionary()
    {
-      // don't load the same dictionary again
       final String language = userPrefs_.spellingDictionaryLanguage().getValue();
-      if (typoLoaded_ && loadedDict_ == language)
+
+      if (!userPrefs_.realTimeSpellchecking().getValue() || typoLoaded_ && loadedDict_.equals(language))
          return;
-      
+
+      // See canRealtimeSpellcheckDict comment, temporary stop gap
+      // Disable real time spellchecking if this dictionary is incompatible
+      // with Typo.js. Final invariant check to ensure that we never try to
+      // load a blacklisted dictionary.
+      if (!canRealtimeSpellcheckDict(language))
+      {
+         userPrefs_.realTimeSpellchecking().setGlobalValue(false);
```

Comment:
I think we should avoid committing a pref change here as it's extremely implicit -- prefs should generally only change due to a user gesture that indicates intent since users can now diff their prefs. Would just bailing from the check do the job here?

Reply:
That should be fine. It's a little weird from the code side because the `LintManager` and other relevant bits will proceed as if real time checking is on but if we always no-op out of the calls I don't see the harm as the user will see the disabled message when they check options.